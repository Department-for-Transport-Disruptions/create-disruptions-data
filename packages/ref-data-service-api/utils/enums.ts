export enum RefVehicleMode {
    bus = "bus",
    tram = "tram",
    coach = "coach",
    ferry = "ferry",
    underground = "underground",
    rail = "rail",
    metro = "metro",
    blank = "",
}
