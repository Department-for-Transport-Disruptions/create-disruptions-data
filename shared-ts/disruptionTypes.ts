import { z } from "zod";
import {
    consequenceOperatorsSchema,
    consequenceSchema,
    disruptionInfoSchema,
    disruptionSchema,
    networkConsequenceSchema,
    operatorConsequenceSchema,
    serviceSchema,
    servicesConsequenceSchema,
    stopSchema,
    stopsConsequenceSchema,
    validitySchema,
} from "./disruptionTypes.zod";

export type Disruption = z.infer<typeof disruptionSchema>;
export type DisruptionInfo = z.infer<typeof disruptionInfoSchema>;

export type Validity = z.infer<typeof validitySchema>;

export type Consequence = z.infer<typeof consequenceSchema>;
export type ConsequenceOperators = z.infer<typeof consequenceOperatorsSchema>;
export type Stop = z.infer<typeof stopSchema>;
export type Service = z.infer<typeof serviceSchema>;

export type NetworkConsequence = z.infer<typeof networkConsequenceSchema>;
export type OperatorConsequence = z.infer<typeof operatorConsequenceSchema>;
export type StopsConsequence = z.infer<typeof stopsConsequenceSchema>;
export type ServicesConsequence = z.infer<typeof servicesConsequenceSchema>;

export const routesSchema = z.object({
    inbound: z.array(stopSchema.partial()),
    outbound: z.array(stopSchema.partial()),
});

export type Routes = z.infer<typeof routesSchema>;
