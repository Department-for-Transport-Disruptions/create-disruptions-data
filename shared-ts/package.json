{
    "name": "@create-disruptions-data/shared-ts",
    "version": "0.1.0",
    "type": "module",
    "scripts": {
        "generate-zod": "npx ts-to-zod ./siriTypes.ts ./siriTypes.zod.ts",
        "kysely": "kysely"
    },
    "dependencies": {
        "@aws-sdk/client-cognito-identity-provider": "^3.682.0",
        "@aws-sdk/client-s3": "^3.682.0",
        "@aws-sdk/client-ssm": "^3.682.0",
        "dayjs": "^1.11.7",
        "kysely": "^0.27.4",
        "lambda-log": "^3.1.0",
        "node-fetch": "^3.3.2",
        "pg": "^8.13.1"
    },
    "devDependencies": {
        "@types/lambda-log": "^2.2.1",
        "@types/pg": "^8.11.10",
        "dotenv": "^16.4.5",
        "kysely-ctl": "^0.9.0",
        "vitest": "^3.0.9"
    }
}
