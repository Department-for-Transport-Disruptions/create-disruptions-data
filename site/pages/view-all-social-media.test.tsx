import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { SocialMediaPostStatus } from "@create-disruptions-data/shared-ts/enums";
import { SocialMediaPost } from "../schemas/social-media.schema";
import ViewAllSocialMedia from "./view-all-social-media.page";

const mockPosts: SocialMediaPost[] = [
    {
        disruptionId: "2cf79629-5c9d-4bc2-8158-7ac68a9759e8",
        messageContent: "a post",
        socialAccount: "25858639",
        hootsuiteProfile: "138196022",
        publishDate: "28/06/2023",
        publishTime: "1700",
        socialMediaPostIndex: 0,
        status: SocialMediaPostStatus.successful,
        accountType: "Hootsuite",
    },
    {
        disruptionId: "2cf79629-5c9d-4bc2-8158-7ac68a9759e8",
        messageContent: "test 12345",
        socialAccount: "25858639",
        hootsuiteProfile: "138196022",
        publishDate: "28/06/2023",
        publishTime: "1800",
        socialMediaPostIndex: 3,
        image: {
            filepath: "/var/folders/85/5c4z0mfs49n56jgsyk737gf80000gp/T/ebe48640bd177bcf210246300",
            mimetype: "image/png",
            size: 70872,
            key: "242ff2b2-19a0-421f-976f-22905262ebda/75164cb7-0a66-420b-8fdc-184ff52cfa71/3.png",
            originalFilename: "1200px-SNice.svg.png",
            url: "https://cdd-image-bucket-teststage.s3.eu-west-2.amazonaws.com/12345",
        },
        status: SocialMediaPostStatus.successful,
        accountType: "Hootsuite",
    },
    {
        disruptionId: "5fb739a8-e5f3-4d3c-b828-f6557c23e834",
        messageContent: "oh no",
        socialAccount: "25858639",
        hootsuiteProfile: "138196022",
        publishDate: "04/07/2023",
        publishTime: "1900",
        socialMediaPostIndex: 0,
        image: {
            filepath: "/var/folders/85/5c4z0mfs49n56jgsyk737gf80000gp/T/d15e6c3b2c83ce2a733ee9701",
            mimetype: "image/png",
            size: 70872,
            key: "242ff2b2-19a0-421f-976f-22905262ebda/5fb739a8-e5f3-4d3c-b828-f6557c23e834/0.png",
            originalFilename: "1200px-SNice.svg.png",
            url: "https://cdd-image-bucket-teststage.s3.eu-west-2.amazonaws.com/12345",
        },
        status: SocialMediaPostStatus.successful,
        accountType: "Hootsuite",
    },
    {
        disruptionId: "640b8b7a-14b8-4f67-9cc3-7f5096a9403c",
        messageContent: "a post",
        socialAccount: "25858639",
        hootsuiteProfile: "138196022",
        publishDate: "28/06/2023",
        publishTime: "1700",
        socialMediaPostIndex: 0,
        status: SocialMediaPostStatus.rejected,
        accountType: "Hootsuite",
    },
    {
        disruptionId: "640b8b7a-14b8-4f67-9cc3-7f8156a9403c",
        messageContent: "twitter post",
        socialAccount: "25858639",
        socialMediaPostIndex: 1,
        status: SocialMediaPostStatus.successful,
        accountType: "Twitter",
    },
];

describe("ViewAllSocialMedia", () => {
    it("should render correctly when social media posts are present", () => {
        const { asFragment } = render(<ViewAllSocialMedia socialMediaPosts={mockPosts} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render correctly when social media posts are not present", () => {
        const { asFragment } = render(<ViewAllSocialMedia socialMediaPosts={[]} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
