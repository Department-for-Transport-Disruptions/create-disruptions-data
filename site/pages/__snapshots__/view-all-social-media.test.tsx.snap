// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ViewAllSocialMedia > should render correctly when social media posts are not present 1`] = `
<DocumentFragment>
  <div
    class="app-phase-banner__wrapper"
  >
    <div
      class="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
    >
      <p
        class="govuk-phase-banner__content"
      >
        <strong
          class="govuk-tag govuk-phase-banner__content__tag"
        >
          beta
        </strong>
        <span
          class="govuk-phase-banner__text"
        >
          This is a new service – your 
          <a
            class="govuk-link"
            href="/feedback"
            id="feedback-link"
          >
            feedback
          </a>
           will help us to improve it.
        </span>
      </p>
    </div>
  </div>
  <div
    class="govuk-width-container"
  >
    <button
      class="govuk-back-link mb-7"
      type="button"
    >
      Back
    </button>
    <main
      class="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        class="govuk-heading-xl"
      >
        Social media posts
      </h1>
      <div
        aria-labelledby="govuk-notification-banner-title"
        class="govuk-notification-banner"
        data-module="govuk-notification-banner"
        role="region"
      >
        <div
          class="govuk-notification-banner__header"
        >
          <h2
            class="govuk-notification-banner__title"
            id="govuk-notification-banner-title"
          >
            Important
          </h2>
        </div>
        <div
          class="govuk-notification-banner__content"
        >
          <p
            class="govuk-notification-banner__heading"
          >
            No live or upcoming disruptions currently have an associated social media message
          </p>
        </div>
      </div>
      <a
        class="govuk-button"
        href="/dashboard"
      >
        Return to dashboard
      </a>
    </main>
    <div>
      <h2
        class="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        class="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link: 
        <a
          class="govuk-link govuk-!-font-size-19"
          href="/contact"
        >
          Contact us
        </a>
      </p>
    </div>
  </div>
  <footer
    class="govuk-footer"
    role="contentinfo"
  >
    <div
      class="govuk-width-container"
    >
      <div
        class="govuk-footer__meta"
      >
        <div
          class="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            class="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            class="govuk-footer__inline-list"
          >
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Help
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/cookies"
              >
                Cookies
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Contact
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
              >
                Terms and conditions
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/privacy"
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            class="govuk-footer__body"
          >
            Built by the 
            <a
              class="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            class="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            class="govuk-footer__licence-description"
          >
            All content is available under the 
            <a
              class="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          class="govuk-footer__meta-item"
        >
          <a
            class="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
          >
            © Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`ViewAllSocialMedia > should render correctly when social media posts are present 1`] = `
<DocumentFragment>
  <div
    class="app-phase-banner__wrapper"
  >
    <div
      class="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
    >
      <p
        class="govuk-phase-banner__content"
      >
        <strong
          class="govuk-tag govuk-phase-banner__content__tag"
        >
          beta
        </strong>
        <span
          class="govuk-phase-banner__text"
        >
          This is a new service – your 
          <a
            class="govuk-link"
            href="/feedback"
            id="feedback-link"
          >
            feedback
          </a>
           will help us to improve it.
        </span>
      </p>
    </div>
  </div>
  <div
    class="govuk-width-container"
  >
    <button
      class="govuk-back-link mb-7"
      type="button"
    >
      Back
    </button>
    <main
      class="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        class="govuk-heading-xl"
      >
        Social media posts
      </h1>
      <div
        class="govuk-accordion"
        data-module="govuk-accordion"
        id="accordion-default"
      >
        <div
          class="govuk-accordion__section"
        >
          <div
            class="govuk-accordion__section-header"
          >
            <h2
              class="govuk-accordion__section-heading"
            >
              <span
                class="govuk-accordion__section-button"
                id="accordion-default-heading-1"
              >
                Social media post 1
              </span>
            </h2>
          </div>
          <div
            aria-labelledby="accordion-default-heading-1"
            class="govuk-accordion__section-content"
            id="accordion-default-content-1"
          >
            <table
              class="govuk-table"
            >
              <thead
                class="govuk-table__head"
              >
                <tr
                  class="govuk-table__row"
                />
              </thead>
              <tbody
                class="govuk-table__body"
              >
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Message to appear
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    a post
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Image
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    No image uploaded
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Publish date
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    28/06/2023
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Publish time
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    1700
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Account name
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    25858639
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    HootSuite profile
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    138196022
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Nextdoor area boundaries
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    N/A
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Status
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    Successful
                  </td>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="govuk-accordion__section"
        >
          <div
            class="govuk-accordion__section-header"
          >
            <h2
              class="govuk-accordion__section-heading"
            >
              <span
                class="govuk-accordion__section-button"
                id="accordion-default-heading-2"
              >
                Social media post 2
              </span>
            </h2>
          </div>
          <div
            aria-labelledby="accordion-default-heading-2"
            class="govuk-accordion__section-content"
            id="accordion-default-content-2"
          >
            <table
              class="govuk-table"
            >
              <thead
                class="govuk-table__head"
              >
                <tr
                  class="govuk-table__row"
                />
              </thead>
              <tbody
                class="govuk-table__body"
              >
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Message to appear
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    test 12345
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Image
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    <a
                      class="govuk-link text-govBlue"
                      href="https://cdd-image-bucket-teststage.s3.eu-west-2.amazonaws.com/12345"
                    >
                      1200px-SNice.svg.png
                    </a>
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Publish date
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    28/06/2023
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Publish time
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    1800
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Account name
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    25858639
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    HootSuite profile
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    138196022
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Nextdoor area boundaries
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    N/A
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Status
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    Successful
                  </td>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="govuk-accordion__section"
        >
          <div
            class="govuk-accordion__section-header"
          >
            <h2
              class="govuk-accordion__section-heading"
            >
              <span
                class="govuk-accordion__section-button"
                id="accordion-default-heading-3"
              >
                Social media post 3
              </span>
            </h2>
          </div>
          <div
            aria-labelledby="accordion-default-heading-3"
            class="govuk-accordion__section-content"
            id="accordion-default-content-3"
          >
            <table
              class="govuk-table"
            >
              <thead
                class="govuk-table__head"
              >
                <tr
                  class="govuk-table__row"
                />
              </thead>
              <tbody
                class="govuk-table__body"
              >
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Message to appear
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    oh no
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Image
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    <a
                      class="govuk-link text-govBlue"
                      href="https://cdd-image-bucket-teststage.s3.eu-west-2.amazonaws.com/12345"
                    >
                      1200px-SNice.svg.png
                    </a>
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Publish date
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    04/07/2023
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Publish time
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    1900
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Account name
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    25858639
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    HootSuite profile
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    138196022
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Nextdoor area boundaries
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    N/A
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Status
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    Successful
                  </td>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="govuk-accordion__section"
        >
          <div
            class="govuk-accordion__section-header"
          >
            <h2
              class="govuk-accordion__section-heading"
            >
              <span
                class="govuk-accordion__section-button"
                id="accordion-default-heading-4"
              >
                Social media post 4
              </span>
            </h2>
          </div>
          <div
            aria-labelledby="accordion-default-heading-4"
            class="govuk-accordion__section-content"
            id="accordion-default-content-4"
          >
            <table
              class="govuk-table"
            >
              <thead
                class="govuk-table__head"
              >
                <tr
                  class="govuk-table__row"
                />
              </thead>
              <tbody
                class="govuk-table__body"
              >
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Message to appear
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    a post
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Image
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    No image uploaded
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Publish date
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    28/06/2023
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Publish time
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    1700
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Account name
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    25858639
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    HootSuite profile
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    138196022
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Nextdoor area boundaries
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    N/A
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Status
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    Rejected
                  </td>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="govuk-accordion__section"
        >
          <div
            class="govuk-accordion__section-header"
          >
            <h2
              class="govuk-accordion__section-heading"
            >
              <span
                class="govuk-accordion__section-button"
                id="accordion-default-heading-5"
              >
                Social media post 5
              </span>
            </h2>
          </div>
          <div
            aria-labelledby="accordion-default-heading-5"
            class="govuk-accordion__section-content"
            id="accordion-default-content-5"
          >
            <table
              class="govuk-table"
            >
              <thead
                class="govuk-table__head"
              >
                <tr
                  class="govuk-table__row"
                />
              </thead>
              <tbody
                class="govuk-table__body"
              >
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Message to appear
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    twitter post
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Image
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    No image uploaded
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Publish date
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    N/A
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Publish time
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    N/A
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Account name
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    25858639
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    HootSuite profile
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    N/A
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Nextdoor area boundaries
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    N/A
                  </td>
                </tr>
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Status
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  >
                    Successful
                  </td>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <a
        class="govuk-button"
        href="/dashboard"
      >
        Return to dashboard
      </a>
    </main>
    <div>
      <h2
        class="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        class="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link: 
        <a
          class="govuk-link govuk-!-font-size-19"
          href="/contact"
        >
          Contact us
        </a>
      </p>
    </div>
  </div>
  <footer
    class="govuk-footer"
    role="contentinfo"
  >
    <div
      class="govuk-width-container"
    >
      <div
        class="govuk-footer__meta"
      >
        <div
          class="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            class="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            class="govuk-footer__inline-list"
          >
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Help
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/cookies"
              >
                Cookies
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Contact
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
              >
                Terms and conditions
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/privacy"
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            class="govuk-footer__body"
          >
            Built by the 
            <a
              class="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            class="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            class="govuk-footer__licence-description"
          >
            All content is available under the 
            <a
              class="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          class="govuk-footer__meta-item"
        >
          <a
            class="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
          >
            © Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
